stacks/shared-ecs/asg.yml

```diff
@@ -99,6 +99,333 @@ Resources:
+                logger.debug("message autoscaling %s", message["LifecycleTransition"])
+                if (
+                    message["LifecycleTransition"].find("autoscaling:EC2_INSTANCE_TERMINATING")
+                    > -1
+                ):
+
+                    # Get lifecycle hook name
+                    lifecycleHookName = message["LifecycleHookName"]
+                    logger.debug("Setting lifecycle hook name %s ", lifecycleHookName)
+
+                    # Check if there are any tasks running on the instance
+                    tasksRunning, tmpMsgAppend = checkContainerInstanceTaskStatus(Ec2InstanceId)
+                    logger.debug("Returned values received: %s ", tasksRunning)
+                    if tmpMsgAppend is not None:
+                        message.update(tmpMsgAppend)
+
+                    # If tasks are still running...
+                    if tasksRunning == 1:
+                        logger.info("SQS delayed callback to %s...", TopicArn)
+                        time.sleep(5)
+                        msgResponse = publishToSNS(message, TopicArn)
+                        logger.debug(
+                            "msgResponse %s and time is %s", msgResponse, datetime.datetime
+                        )
+                    # If tasks are NOT running...
+                    elif tasksRunning == 0:
+                        # completeHook = 1
+                        logger.debug(
+                            (
+                                "Setting lifecycle to complete;No tasks are running "
+                                "on instance, completing lifecycle action...."
+                            )
+                        )
+
+                        try:
+                            response = asgClient.complete_lifecycle_action(
+                                LifecycleHookName=lifecycleHookName,
+                                AutoScalingGroupName=asgGroupName,
+                                LifecycleActionResult="CONTINUE",
+                                InstanceId=Ec2InstanceId,
+                            )
+                            logger.info(
+                                "Response received from complete_lifecycle_action %s", response
+                            )
+                            logger.info("Completedlifecycle hook action")
+                        except Exception as e:
+                            print(str(e))
+      MemorySize: 256
+      Policies:
+        - Statement:
+            - Action:
+                - ec2:DescribeInstances
+                - ec2:DescribeInstanceAttribute
+                - ec2:DescribeInstanceStatus
+                - ec2:DescribeHosts
+                - ecs:DescribeContainerInstances
+                - ecs:DescribeTasks
+                - ecs:ListContainerInstances
+                - ecs:ListTasks
+                - sns:ListSubscriptions
+              Effect: Allow
+              Resource: "*"
```

Comment:
Some of these seem a bit permissive?  At least they're read permissions.  May want to lock these down a bit though, once I prove this (still) works.